torch/fx/passes/fake_tensor_prop.py

```diff
@@ -23,43 +23,35 @@ class FakeTensorProp(torch.fx.Interpreter):
          module (GraphModule): The module to be executed
          mode (Optional[FakeTensorMode]): The dispatch mode used to execute computation indicated by each FX Node.
     """
-    def __init__(self, module: torch.fx.GraphModule, mode: Optional[FakeTensorMode] = None, *, check_consistency: bool = True):
+    def __init__(self, module: torch.fx.GraphModule, mode: Optional[FakeTensorMode] = None):
         super().__init__(module)
         if mode is None:
             mode = FakeTensorMode()
         self._mode = mode
-        self.check_consistency = check_consistency
 
     def run_node(self, n: Node):
+        from torch.fx.experimental.symbolic_shapes import rebind_unbacked, compute_unbacked_bindings
 
         result = super().run_node(n)
+        rebind_unbacked(self._mode.shape_env, n, result)
```

Comment:
This is basically a revert of https://github.com/pytorch/pytorch/pull/124059, except for this line which is new.

Reply:
Just to make sure I understand. This revert comes from the fact that we don't need to do the check because we are effectively making sure that the old and the new are the same in `rebind_unbacked`. Is that correct?

Reply:
Yup. And the old code was doing "too much": it was testing equality of every output, even if there are no unbacked symints. @IvanKobzarev was doing some profiling and these equality tests were expensive. So while they might have some utility in a debugging mode, for regular mode we should just do the bare minimum we need, which is rebinding only.